pkg/arm/v15/data/master-startup.sh

```diff
@@ -91,6 +87,21 @@ done
 set +x
 export SASURI='{{ .Config.MasterStartupSASURI }}'
 set -x
+
+# run the startup --init to bootstrap DNS setup
+docker run --privileged --rm --network host -v /:/host:z -e SASURI {{ .Config.Images.Startup }} startup --init-network
+
+# restart network manager to pick up new host settings in the dhclient
+/bin/systemctl restart NetworkManager
+# set the /etc/resolv.conf
+/etc/NetworkManager/dispatcher.d/99-origin-dns.sh
+# restart dnsmasq to get the new settings in /etc/dnsmasq.conf
+/bin/systemctl restart dnsmasq.service
```

Comment:
I thought that a NetworkManager restart would run 99-origin-dns.sh, and maybe even that 99-origin-dns.sh would restart dnsmasq.  Are both of these lines definitely necessary?  Why?

Reply:
What I found during testing(from memory) was the following. 

1. The restart to network manager was required to pick up the dhclient-eth0.conf changes.
1. On restart the network manager would place the dhclient-eth0.conf into /var/lib/NetworkManager/dhclient-eth0.conf which would cause a refresh of the lease file.
1. This would cause NetworkManager to update /etc/resolv.conf with the new search domain and the dns servers.  I'm not 100% sure if the 99-origin-dns.sh would rerun.  I had found where the settings for the local dnsmasq's 10. address were not in the resolv.conf.  A rerun of the 99-origin-dns.sh would ensure that the 10. address was in resolv.conf and that the search domains were ordered correctly, cluster.local first, followed by azure domain.
1. Then finally the dnsmasq.service restart that causes dnsmasq to pick up the newly appended search string `server=/<Azure domain>/168.63.129.16`.  In testing, I think that the NetworkManager restart did not pick up the latest settings from dnsmasq.conf.